backend/core/detection.py

```diff
@@ -830,6 +844,33 @@ def mark_outdated(self, hij_key: str, redis_hij_key: str) -> NoReturn:
                 msg, exchange=self.hijack_exchange, routing_key="outdate", priority=1
             )
 
+        def gen_implicit_withdrawal(self, monitor_event: Dict) -> NoReturn:
+            """
+            Checks if a benign BGP update should trigger an implicit withdrawal
+            """
+            # log.debug('{}'.format(monitor_event['key']))
+            prefix = monitor_event["prefix"]
+            peer_asn = monitor_event["peer_asn"]
+            if self.redis.exists("prefix_{}_peer_{}_hijacks".format(prefix, peer_asn)):
+                # generate implicit withdrawal
+                withdraw_msg = {
+                    "service": "implicit-withdrawal",
+                    "type": "W",
+                    "prefix": prefix,
+                    "path": [],
+                    "orig_path": {"triggering_bgp_update": monitor_event},
+                    "communities": [],
+                    "timestamp": monitor_event["timestamp"],
+                    "peer_asn": peer_asn,
+                }
+                key_generator(withdraw_msg)
+                self.producer.publish(
```

Comment:
why do we need to send to the update exchange and not just send the correct messages to the withdraw exchange directly?

Reply:
are these two equivalent, regarding their handling? If yes, I will use the withdraw exchange, no problem.

Reply:
also as I see in the rest of the detection code:
```
elif monitor_event["type"] == "W":
                self.producer.publish(
                    {
                        "prefix": monitor_event["prefix"],
                        "peer_asn": monitor_event["peer_asn"],
                        "timestamp": monitor_event["timestamp"],
                        "key": monitor_event["key"],
                    },
                    exchange=self.update_exchange,
                    routing_key="withdraw",
                    priority=0,
)
```
i.e., we use the update exchange to send withdrawals. This is yet another withdrawal, so it should go over the same pipeline, correct?

Reply:
withdraw routing_key goes directly to the database module while the update routing_key requeues it to the detection module. Does it make sense to go through the detection again?

Reply:
if there are no implications I will send it directly, good call.

Reply:
see slack discussion, sending it directly results to non-registration of the withdrawal to the db and the loss of useful information.

Reply:
resolving as discussed. The implicit withdrawal needs to be handled by the db, and then the detector should be notified and synced with the DB w.r.t. this as with the other withdrawals (current solution, compliant with the initial design logic).